### ISSUE TITLE
Start Segment Replace Request API does not allow empty segmentsTo

### ISSUE THREAD

**Issue by harold-kfuse** [apache/pinot#10430]
Discussed with @Jackie-Jiang offline. He mentioned that the Segment replace api allows M>=0 to N>=0 segments, i.e., I can pass an empty `segmentsTo` list of segments and it should work and will mark the segments in segmentsFrom for deletion). 

However, when I tried giving an empty segmentsTo list, I get a precondition check failure: https://github.com/apache/pinot/blob/7e9ca6a5a4afe0d4e283ac1307c45430e474cbf2/pinot-common/src/main/java/org/apache/pinot/common/restlet/resources/StartReplaceSegmentsRequest.java#L43

Is the preconditions check required?

**Comment by Jackie-Jiang** [comment]
@snleee Can you take a look at this issue? IMO we should allow empty `segmentsTo` to support atomic batch deletion

**Comment by saurabhd336** [comment]
`/segments/{tableName}/delete` already supports atomic batch deletion @Jackie-Jiang ?

**Comment by Jackie-Jiang** [comment]
@saurabhd336 I don't think that API is doing atomic batch delete. It doesn't involve lineage change (segment replace transaction).

**Comment by snleee** [comment]
Thank you @harold-kfuse for reporting this. I think that we should support `[s1, s2, s3] -> []` to allow batch deletion of the segments and it is the valid case. 

Let me file a pr to address this.
